File: 백준/Silver/2503. 숫자 야구/숫자 야구.java
import java.io.*;
import java.util.StringTokenizer;
import java.util.ArrayList;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        ArrayList<String> nums = new ArrayList<>();
        // 숫자 만들기
        for (int i = 1; i <= 9; i++) {
            for (int j = 1; j <= 9; j++) {
                if (i != j) {
                    for (int k = 1; k <= 9; k++) {
                        if (j != k && k != i) {
                            int tmp = i * 100 + j * 10 + k;
                            nums.add(String.valueOf(tmp));
                        }
                    }
                }
            }
        }

        StringTokenizer st;
        int[] numCnt = new int[nums.size()];
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String num = st.nextToken();
            int strike = Integer.parseInt(st.nextToken());
            int ball = Integer.parseInt(st.nextToken());

            for (int j = 0; j < nums.size(); j++) {
                String n2 = nums.get(j);
                int s = 0;
                int b = 0;
                for (int k = 0; k < 3; k++) {
                    if (n2.contains(String.valueOf(num.charAt(k)))) {
                        b++;
                    }
                    if (num.charAt(k) == n2.charAt(k)) {
                        s++;
                        b--;
                    }

                }

                if (s == strike && b == ball) {
                    numCnt[j]++;
                }
            }
        }
        int answer = 0;
        for (int i = 0; i < numCnt.length; i++) {
            if (numCnt[i] == N) {
                answer++;
            }
        }
        System.out.println(answer);
    }
}

File: 백준/Silver/2503. 숫자 야구/숫자 야구.py
nums = []
for i in range(1, 10):
    for j in range(1, 10):
        if i == j:
            continue
        for k in range(1, 10):
            if i != k and j != k:
                tmp = i * 100 + j * 10 + k
                nums.append(str(tmp))

N = int(input())
num_cnt = [0] * len(nums)

for _ in range(N):
    num, strike, ball = map(int, input().split())
    num = str(num)

    for i in range(len(nums)):
        n2 = nums[i]
        s = 0
        b = 0

        for j in range(3):
            if num[j] in n2:
                b += 1

            if num[j] == n2[j]:
                s += 1
                b -= 1
        if s == strike and b == ball:
            num_cnt[i] += 1

answer = 0
for i in range(len(num_cnt)):
    if num_cnt[i] == N:
        answer += 1

print(answer)